Craig Leres identified a problem that makes fetchmail 6.3.2 (only this
version) crash if the .netrc file does not contain a password for a
particular account.

This patch is mostly Craig Leres' work has been committed to the
SVN repository and should be applied to fetchmail 6.3.2 on all sites
that plan to use netrc files:

Index: netrc.c
===================================================================
--- netrc.c	(Revision 4683)
+++ netrc.c	(Revision 4684)
@@ -314,8 +314,10 @@
 free_netrc(netrc_entry *a) {
     while(a) {
 	netrc_entry *n = a->next;
-	memset(a->password, 0x55, strlen(a->password));
-	xfree(a->password);
+	if (a->password != NULL) {
+		memset(a->password, 0x55, strlen(a->password));
+		free(a->password);
+	}
 	xfree(a->login);
 	xfree(a->host);
 	xfree(a);

Sorry for the inconvenience.  -- Matthias Andree, 2006-01-30
